kiwi/firmware.py

```diff
@@ -28,22 +28,13 @@
 
 class FirmWare(object):
     """
-    Implements firmware specific image information
+    **Implements firmware specific methods**
 
     According to the selected firmware some parameters in a disk
     image changes. This class provides methods to provide firmware
     dependant information
 
-    Attributes
-
-    * :attr:`arch`
-        machine architecture
-
-    * :attr:`zipl_target_type`
-        XML configured zipl target type
-
-    * :attr:`firmware`
-        XML configured firmware name
+    * :param object xml_state: instance of XMLState
```

Comment:
Hmn, I think, this is a contradiction. Shouldn't it be something like this? Also added `:class:` directive:

```
   :param XMLState xml_state: instance of :class:`XMLState`
```

Reply:
Hmm, I always used __:param object argument_name: instance of Class__  to say argument_name is a python object of the class Class

where is this a contradiction ? and if so there are many many cases where I used it oh oh

Reply:
> Hmm, I always used :param object argument_name: instance of Class to say argument_name is a python object of the class Class

According to the Sphinx documentation, the synopsis for a `:param:` directive is this:

```
:param [TYPE] VARNAME: DESCRIPTION
```

In some cases, you can omit the `type` and write this:

```
:param VARNAME: DESCRIPTION
:type VARNAME: Type1, Type2
```

This can be used if the argument can have different datatypes (for example, a `str` or `bytes`).

In the above example, you used `object` for the datatype. Although, technically it is correct, it is **very** broad. Everything is an object in Python. :wink: It doesn't give the reader additional information.  If a developer reads this information, he should know what type the function/method is expected and IMHO `object` is not correct.

In your case, it's not only one object, but a specific object: an instance of the class `XMLState`.

> where is this a contradiction ? and if so there are many many cases where I used it oh oh

Ok, maybe contradiction was the wrong word; let's agree on that it's ambiguous. :wink:

If it causes too much trouble on your side, lets take the current state for now.